### PR TITLE
Add rspec-specific GitHub workflow

### PR DESCRIPTION
**Pull request by zzaakiirr**
### What does this PR do and why?

This PR adds new `rspec-spicific` GitHub workflow so that job can be triggered manually to run specific tests

### Does it work?

Example of manually triggered job with passing `spec/command/config_spec.rb` as `spec_paths` argument - https://github.com/zzaakiirr/control-plane-flow/actions/runs/10020011552/job/27697199967

Screenshot 2024-07-20 at 15 07 02

Note, that "Setup Control Plane tools" step is intentionally skipped and tests fail because of that

## Summary by CodeRabbit

- **New Features**
	- Introduced a new workflow for running specific RSpec tests, allowing users to manually specify test files or directories.
	- Enhanced configurability of the shared workflow by making `test_tag` optional and adding `spec_paths` for more targeted test executions.

- **Improvements**
	- Increased flexibility in test execution by allowing conditions on the inclusion of tags and specifying paths for RSpec tests.
	- Standardized naming conventions for workflow parameters, improving consistency and readability.

### CODE REVIEW

File: .github/workflows/rspec-specific.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      spec_paths:
+        description: "Test files or directories that should be run"
+        required: true
+
+jobs:
+  rspec-specific-tests:
```
Comment (rafaelgomesxyz) [inline .github/workflows/rspec-specific.yml:11]: Could be just `rspec-specific`.

File: .github/workflows/rspec-shared.yml [hunk 2]
```diff
@@ -10,7 +10,10 @@ on:
         required: true
         type: string
       test-tag:
-        required: true
+        required: false
+        type: string
+      spec_paths:
```
Comment (rafaelgomesxyz) [inline .github/workflows/rspec-shared.yml:15]: We're mixing `-` and `_`, and I just realized that `_` is more appropriate. Can you please change it to use `_` for all inputs in all workflows?

I know that's a bit out of scope for this PR, but we might as well fix it here.

File: .github/workflows/rspec-shared.yml [hunk 3]
```diff
@@ -41,7 +44,7 @@ jobs:
           cpln profile create default --token $CPLN_TOKEN_CI --org $CPLN_ORG
           cpln image docker-login
       - name: Run tests
-        run: bundle exec rspec --format documentation --tag ${{ inputs.test-tag }}
+        run: bundle exec rspec --format documentation ${{ inputs.test-tag && format('--tag {0}', inputs.test-tag) }} ${{ inputs.spec_paths }}
```
Comment (rafaelgomesxyz) [inline .github/workflows/rspec-shared.yml:47]: Where does `format` come from?

**Review by rafaelgomesxyz — COMMENTED** [review]
Left a few comments.

By the way, what do you mean by "Setup Control Plane tools step is intentionally skipped"?

File: .github/workflows/rspec-shared.yml (hunk 3 above)
Comment (zzaakiirr) [reply .github/workflows/rspec-shared.yml:47]: It's a built-in function which can be used in expressions - https://docs.github.com/en/actions/learn-github-actions/expressions#format

**Comment by zzaakiirr** [comment]
> By the way, what do you mean by "Setup Control Plane tools step is intentionally skipped"?

I removed "Setup Control Plane tools step is intentionally skipped" from `rspec-shared.yml` because didn't want to bother with setting up `CPLN_ORG` and `CPLN_TOKEN_CI` variables and just wanted to check if `bundle exec rspec` runs with correct arguments - https://github.com/zzaakiirr/control-plane-flow/commit/96bc5c3ecb8d1203105b290088307b1fca8ba202

**Review by rafaelgomesxyz — APPROVED** [review]
Thanks for this!

**Pull request merged**